### PR TITLE
[1.26] Add test to complete the coverage of urllib3.util.ssl_match_hostname.match_hostname

### PR DESCRIPTION
**Pull request by hramezani**
Ref: https://github.com/urllib3/urllib3/issues/2543
This path Covers https://github.com/urllib3/urllib3/blob/b1f60e44d43b13e5272d5b6003f125af9c25c8ad/src/urllib3/util/ssl_match_hostname.py#L150

### CODE REVIEW

File: test/with_dummyserver/test_https.py [hunk 1]
```diff
@@ -858,6 +858,26 @@ def test_warning_for_certs_without_a_san(self, no_san_server):
                 assert r.status == 200
                 assert warn.called
 
+    def test_common_name_without_san_with_different_common_name(
+        self, no_san_server_with_different_commmon_name
+    ):
+        ctx = util.ssl_.create_urllib3_context()
+        try:
```
Comment (sethmlarson) [inline test/with_dummyserver/test_https.py:865]: Since this is 1.26.x I think we're always checking commonName?

**Review by sethmlarson — COMMENTED** [review]
Thanks for this! One comment:

File: test/with_dummyserver/test_https.py (hunk 1 above)
Comment (hramezani) [reply test/with_dummyserver/test_https.py]: Yes, You are right. Fixed!

**Review by sethmlarson — APPROVED** [review]

**Comment by sethmlarson** [comment]
Silly me, we're not 100% coverage on this branch so auto-merge can't work! Let me merge manually.

**Pull request merged**